src/bci_build/package/golang/README.md.j2

```diff
@@ -58,26 +86,69 @@ To run the application tests inside a container, use the following command:
 $ podman run --rm -v "$PWD":/app:Z -w /app {{ image.pretty_reference }} go test -v
 ```
 
-**Note:** The Golang image should be used as a build environment. For runtime, self-contained Go binaries should use a `scratch` image and for applications that require external dependencies use the `bci-base` image.
+### Building from SLE BCI
 
-## Additional tools
+The Go image should be used as a build environment. That means
+the compiler does not need to be shipped as part of the images that are
+distributed. Instead, it is recommended that the Go image is used as the
+builder image only.
 
-The following additional tools are included in the image:
+The [SLE BCI General Purpose Base Containers](https://opensource.suse.com/bci-docs/documentation/general-purpose-bci/)
+images offer four different options here, depending on your exact requirements.
 
-- go{{ image.version }}-race
-- make
-- git-core
+```Dockerfile
+# Build the application using the {{ image.pretty_name }} Container Image
+FROM {{ image.pretty_reference }} as build
 
-{% if image.version.endswith('-openssl') %}
-## FIPS 140
+WORKDIR /app
 
-The image includes a FIPS 140 enabled Go toolchain using OpenSSL.
+# pre-copy/cache go.mod for pre-downloading dependencies and only
+# redownloading them in subsequent builds if they change
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
 
-To restrict all TLS configuration to FIPS-approved settings, add the following line:
+COPY . ./
+
+# Make sure to build the application with CGO disabled.
+# This will force Go to use some Go implementations of code
+# rather than those normally supplied by the host operating system.
+# You need this for scratch images as those supporting libraries
+# are not available.
+RUN CGO_ENABLED=0 go build -o /hello
+
+# Bundle the application into a scratch image
+FROM registry.suse.com/bci/bci-micro:15.4
+
+COPY --from=build /hello /hello
+
+CMD ["/hello"]
+```
+
+
+The above example uses the SLE BCI micro image as the deployment image for the resulting application.
+See the [SLE BCI use with Go documentation](https://opensource.suse.com/bci-docs/guides/use-with-golang/)
+for further details.
+
+{%- if image.version.endswith('-openssl') %}
+
+## FIPS 140-3
+
+To restrict all TLS configuration to FIPS-approved settings, add
+the following line:
 
 ```go
 import _ "crypto/tls/fipsonly"
 ```
-{% endif %}
+
+{%- endif %}
+
+
+## Additional tools
+
+The following tools are also included in the image:
+
+- go{{ image.version }}-race
+- make
+- git-core
```

Comment:
How about this? Not everyone is familiar with our git packaging
```suggestion
- git (provided by the package `git-core`)
```

Reply:
I'd just delete all of these package listings imho (in a separate PR)